fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java

```diff
@@ -617,6 +629,26 @@ public long loadBackupHandler(DataInputStream dis, long checksum, GlobalStateMgr
         LOG.info("finished replay backupHandler from image");
         return checksum;
     }
+
+    /**
+     * will remove finished/cancelled job periodically
+     */
+    private boolean isJobExpired(AbstractJob job, long currentTimeMs) {
+        return (job.isDone() || job.isCancelled())
+                && (currentTimeMs - job.getFinishedTime()) / 1000 > Config.history_job_keep_max_second;
+    }
+
+    public void removeOldJobs() {
+        long currentTimeMs = System.currentTimeMillis();
+        Iterator<Map.Entry<Long, AbstractJob>> iterator = dbIdToBackupOrRestoreJob.entrySet().iterator();
```

Comment:
Is this thread safe? even though dbIdToBackupOrRestoreJob is a concurrent hash map

Reply:
I've added a sequence lock here. PTAL.